### DISCUSSION TITLE
Jekyll Build failing for Github pages site

### DISCUSSION THREAD

**Discussion by way0utwest** [community/community#127006] [Actions] [answered]
### Select Topic Area

Question

### Body

I started receiving build errors today from my github pages site based on Jekyll. I got this error in a CI build:

`Fetching gem metadata from https://rubygems.org/............`
`Fetching gem metadata from [https://rubygems.org/.](https://rubygems.org/)`
`Resolving dependencies...`
`ffi-1.17.0-x86_64-linux-musl requires rubygems version >= 3.3.22, which is`
`incompatible with the current version, 3.0.6`

I hadn't changed anything, and I'm guessing a dependent page changed, or the build agent changed.

My ci.yml just has:
`name: Beautiful Jekyll CI`
`on: [push, pull_request]`
`jobs:`
`  build:`
`    name: Build Jekyll`
`    runs-on: ubuntu-latest`
`    steps:`
`      - uses: actions/checkout@v2`
`      - name: Build the site in the jekyll/builder container`
`        run: |`
`          export JEKYLL_VERSION=3.8`
`          docker run \`
`          -v ${{ github.workspace }}:/srv/jekyll -v ${{ github.workspace }}/_site:/srv/jekyll/_site \`
`          -e PAGES_REPO_NWO=${{ github.repository }} \`
`          jekyll/builder:$JEKYLL_VERSION /bin/bash -c "chmod 777 /srv/jekyll && jekyll build --future"`

Does anyone know why this started failing now?

**Comment by vshunkov** [comment]
Hi, I am facing the same error but in different place. So far I have found that it is related to the [dependency update](https://my.diffend.io/gems/ffi/1.17.0.rc2-x86_64-linux-musl/1.17.0-x86_64-linux-musl) in the latest version, which was released yesterday. Does someone have any findings?

**Comment by way0utwest** [comment]
Or if there is a way to specify an older version of Linux to avoid this error?

**Comment by vshunkov** [comment]
Gemfile.lock should be a right place but in my case ffi-1.17.0 being installed as an dependency for a external library. Btw, I opened a ticket for them https://github.com/ffi/ffi/issues/1103

**Comment by vshunkov** [comment]
ffi explicit lock to the version 1.16.3 fixed the problem on my end.

**Reply by way0utwest** [reply]
Lock where in code? Sorry for the question, but I'm just building on GH and a bit of a jekyll newbie. This has been working for a few years with a basic pipeline and just started failing.

**Reply by vshunkov** [reply, marked as answer]
I am not familiar with Jenkyll specific but from ruby side it is the same, try to create Gemfile.lock inside your project with the content:
Source  “https://rubygems.org”
gem “ffi”,  “= 1.16.3”

**Reply by way0utwest** [reply]
Thanks! That exactly worked.

**Reply by lwasser** [reply]
worked for us as well. we have the same issue with jekyll build right now! All of our ci builds were breaking!

**Comment by idroulia** [comment]
Thanks for this! I was trying to figure out the issue for an hour. Downgrading ffi worked like a charm

**Comment by Vyshnav2255** [comment]
Didn't work for me

Using Bundler version 2.5.6 from Gemfile.lock
Failed during stage 'Install dependencies': dependency_installation script returned non-zero exit code: 1
ERROR:  Error installing bundler:
There are no versions of bundler (= 2.5.6) compatible with your Ruby & RubyGems
bundler requires Ruby version >= 3.0.0. The current ruby version is 2.6.2.47.
Error installing Bundler
Failing build: Failed to install dependencies
